app/src/main/java/io/github/sds100/keymapper/mappings/keymaps/ListKeyMapsUseCase.kt

```diff
@@ -17,23 +19,25 @@ class ListKeyMapsUseCaseImpl(
     private val keyMapRepository: KeyMapRepository,
     private val backupManager: BackupManager,
     displayKeyMapUseCase: DisplayKeyMapUseCase,
+    private val observeKeyMapsSorterUseCase: ObserveKeyMapsSorterUseCase,
 ) : ListKeyMapsUseCase,
     DisplayKeyMapUseCase by displayKeyMapUseCase {
 
     override val keyMapList: Flow<State<List<KeyMap>>> = channelFlow {
         send(State.Loading)
 
-        keyMapRepository.keyMapList.collectLatest { keyMapEntitiesState ->
-            send(State.Loading)
-
+        combine(
+            keyMapRepository.keyMapList,
+            observeKeyMapsSorterUseCase(),
+        ) { keyMapEntitiesState, sorter ->
+            keyMapEntitiesState.mapData { keyMapEntities ->
+                keyMapEntities
+                    .map { KeyMapEntityMapper.fromEntity(it) }
+                    .sortedWith(sorter)
```

Comment:
I am not sure if I am missing something.

I can’t just reverse the whole list because it wouldn’t make sense with the field priorities. Instead, I need to reverse specific comparators to get the sorting right.

---

I don't know how to move sorting in a clean way because listing depends entirely on sorting.

Currently it works like this:
1. User sets order 
2. New sorter is created and notifies observers (`ListKeyMapsUseCase`)
3. `ListKeyMapsUseCase` updates the list

If I create a method that takes a list and returns sorted lists, there’s no way for the UI to update itself.

Reply:
Ahh, yes. Lets keep the reverse field then. Can you append the comment above the `reversed` field with what you said so we don't forget in the future.

Reply:
Did it myself so we can wrap this up quickly :)